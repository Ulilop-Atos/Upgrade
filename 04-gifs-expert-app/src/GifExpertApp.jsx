import PropTypes from 'prop-types';
import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifSearcher from './components/GifSearcher';

const GifExpertApp = (props) => {

    const [categories, setCategories] = useState(['One Punch Man', 'Naruto']);

    return (
        <>
            <h2> GIFExpertApp </h2>
            <AddCategory updateArray={ setCategories } />
            <hr/>
            <ul>
            {
                categories.map( (category, i) => 
                    <GifSearcher 
                    key={`${i}-${category}`}
                    category={category} 
                    /> 
                )
            }
            </ul>
        </>
    );
};

GifExpertApp.propTypes = {
    props: PropTypes.any,
};

export default GifExpertApp;
