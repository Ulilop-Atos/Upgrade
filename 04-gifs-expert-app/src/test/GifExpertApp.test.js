import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Tests for <GifExpertApp />', () => { 
  test('should display properly', () => { 
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should display a list of categories', () => { 
    const categories = ['One Punch Man', 'Naruto', 'Karla'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifSearcher').length).toBe(categories.length);
   })
});