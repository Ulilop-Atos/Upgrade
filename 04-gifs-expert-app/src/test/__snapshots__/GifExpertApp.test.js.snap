// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifExpertApp /> should display a list of categories 1`] = `
<Fragment>
  <h2>
     GIFExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifSearcher
      category="One Punch Man"
      key="0-One Punch Man"
    />
    <GifSearcher
      category="Naruto"
      key="1-Naruto"
    />
    <GifSearcher
      category="Karla"
      key="2-Karla"
    />
  </ul>
</Fragment>
`;

exports[`Tests for <GifExpertApp /> should display properly 1`] = `
<Fragment>
  <h2>
     GIFExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifSearcher
      category="One Punch Man"
      key="0-One Punch Man"
    />
    <GifSearcher
      category="Naruto"
      key="1-Naruto"
    />
  </ul>
</Fragment>
`;
